### PR TITLE
add a metric for a metric by family

### PR DESCRIPTION
**Pull request by djeebus**
also some refactoring to remove web code from biz logic

### CODE REVIEW

File: publicRouter.go [hunk 1]
```diff
@@ -25,10 +30,41 @@ func setupAPIRouter(corsDomain string, agg *aggregate, promConfig metrics.Config
...
+
+func handleRender(a *aggregate) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		contentType := expfmt.Negotiate(c.Request.Header)
+		c.Header("Content-Type", string(contentType))
+		enc := expfmt.NewEncoder(c.Writer, contentType)
+
+		a.render(enc)
+	}
+}
```
Comment (mplachter) [inline publicRouter.go:70]: I found this API change confusing. Not sure if there is any benefit in making this change.

I would probably error on the side of not making this change if there is no clear benefit.

Comment (davidwin93) [reply publicRouter.go:70]: I also don't see a benefit in this change. Is there a reason for moving these functions out of aggregate @djeebus ?

File: aggregate.go [hunk 2]
```diff
@@ -29,25 +33,18 @@ type aggregate struct {
 type ignoredLabels []string
 
 type aggregateOptions struct {
-	ignoredLabels     ignoredLabels
-	metricTTLDuration *time.Duration
+	ignoredLabels ignoredLabels
 }
 
-type aggregateOptionsFunc func(a *aggregate)
+type AggregateOptionsFunc func(a *aggregate)
```
Comment (mplachter) [inline aggregate.go:39]: this should stay private any reason on making this public?

**Comment by djeebus** [comment]
punting on this; ended up being just a refactor. not necessary at this point.

**Pull request closed without merging**